src/AppInstallerRepositoryCore/PackageDependenciesValidation.cpp

```diff
@@ -0,0 +1,253 @@
+            std::for_each(
+                packageLatestDependencies.begin(),
+                packageLatestDependencies.end(),
+                [&](std::pair<std::pair<SQLite::rowid_t, Utility::NormalizedString>, SQLite::rowid_t> row)
+                {
+                    auto manifestRowId = index->GetManifestIdByKey(row.first.first, "", "");
+                    auto packageId = index->GetPropertyByManifestId(manifestRowId.value(), PackageVersionProperty::Id);
+                    Dependency dep(DependencyType::Package, packageId.value(), row.first.second);
+                    depList.Add(dep);
+                });
+
+            return depList;
+            });
+
+        graph.BuildGraph();
+
+        if (foundErrors)
+        {
+            THROW_EXCEPTION(ManifestException(std::move(dependenciesError), APPINSTALLER_CLI_ERROR_DEPENDENCIES_VALIDATION_FAILED));
+        }
+
+        if (graph.HasLoop())
+        {
+            std::string error = ManifestError::FoundLoop;
+            dependenciesError.emplace_back(error);
+            THROW_EXCEPTION(ManifestException(std::move(dependenciesError), APPINSTALLER_CLI_ERROR_DEPENDENCIES_VALIDATION_FAILED));
+        }
+
+        return true;
+    }
+
+    bool PackageDependenciesValidation::VerifyDependenciesStructureForManifestDelete(SQLiteIndex* index, const Manifest::Manifest manifest)
+    {
+        auto dependentsSet = index->GetDependentsById(manifest.Id);
+
+        if (!dependentsSet.size())
+        {
+            // all good this manifest is not a dependency of any manifest.
+            return true;
+        }
+
+        std::vector<std::pair<Manifest::Manifest, Utility::Version>> manifestToVersionPair;
+        std::for_each(
+            dependentsSet.begin(),
+            dependentsSet.end(),
+            [&](std::pair<SQLite::rowid_t, Utility::Version> current)
+            {
+                Manifest::Manifest manifest;
+                manifest.Id = index->GetPropertyByManifestId(current.first, PackageVersionProperty::Id).value();
+                manifest.Version = index->GetPropertyByManifestId(current.first, PackageVersionProperty::Version).value();
+
+                manifestToVersionPair.emplace_back(std::make_pair(manifest, current.second));
+            });
+
+        auto packageLatest = GetPackageLatestVersion(index, manifest.Id);
+
+        if (!packageLatest.has_value())
+        {
+            // this is a fatal error, a manifest should exists in the very least(including the current manifest being deleted),
+            // since this is a delete operation. 
+            THROW_HR(APPINSTALLER_CLI_ERROR_MISSING_PACKAGE);
+        }
+
+        if (Utility::Version(manifest.Version) < packageLatest.value().second)
+        {
+            // all good, since it's min version the criteria is still satisfied.
+            return true;
+        }
+
+        auto nextLatestAfterDelete = GetPackageLatestVersion(index, manifest.Id, { packageLatest.value().second });
```

Comment:
>GetPackageLatestVersion [](http://example.com/codeflow?start=37&length=23)

It seems like it would be easier for this to just return all of the manifests and their versions.  Or at least between 0 and 2 of them.

Reply:
And it should probably filter out anything that isn't empty channel for now.

Reply:
GetPackageLatest approach is much consistent with the install flow implementation and it's clearer that the latest version is being selected by just reading the code for manitainability sake. Getting a list of manifest might seem "easier"(I don't see how), it just isn't consistent with the install flow implementation. Btw, GetPackageLatest has been implemented to handle channels.